matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MatpowerReader.java

```diff
@@ -145,17 +162,19 @@ private static void readGenerators(Matrix generators, MatpowerModel model) {
             gen.setStatus(generators.getInt(row, 7));
             gen.setMaximumRealPowerOutput(generators.getDouble(row, 8));
             gen.setMinimumRealPowerOutput(generators.getDouble(row, 9));
-            gen.setPc1(generators.getDouble(row, 10));
-            gen.setPc2(generators.getDouble(row, 11));
-            gen.setQc1Min(generators.getDouble(row, 12));
-            gen.setQc1Max(generators.getDouble(row, 13));
-            gen.setQc2Min(generators.getDouble(row, 14));
-            gen.setQc2Max(generators.getDouble(row, 15));
-            gen.setRampAgc(generators.getDouble(row, 16));
-            gen.setRampTenMinutes(generators.getDouble(row, 17));
-            gen.setRampThirtyMinutes(generators.getDouble(row, 18));
-            gen.setRampQ(generators.getDouble(row, 19));
-            gen.setApf(generators.getDouble(row, 20));
+            if (generatorColumnsToRead > 9) {
```

Comment:
Here, the method will throw an exception in case of a generator table in v1 format (method called with `MATPOWER_V1_GENERATORS_COLUMNS` which equals 10).

But it may be clearer to have this test instead:
```suggestion
            if (generatorColumnsToRead == MATPOWER_V2_GENERATORS_COLUMNS) {
```

because it will be easier to relate it to the possible values for `generatorColumnsToRead`.

Reply:
I tried another design

Reply:
Thanks, it looks good to me.